charts/core/src/XAxis/XAxis.types.ts

```diff
@@ -18,7 +18,21 @@ export interface XAxisProps {
   label?: string;
 
   /**
-   * Unit of the axis to be rendered with value.
+   *
+   * Formatter of axis label, which supports string template and callback function.
+   *
+   * ```ts
+   * // Use string template; template variable is the default label of axis {value}
+   * formatter: '{value} kg'
+   *
+   * // Use callback.
+   * formatter: function (value, index) {
+   *   return value + 'kg';
+   *}
+   * ```
    */
-  unit?: string;
+  formatter?:
+    | string
+    | ((value: string, index: number) => string)
+    | { [key: string]: string };
```

Comment:
What would an object formatter look like?

Reply:
The object formatter is what echarts calls a "cascading template" and only works with time. From their site:
> Sometimes, we wish to use different formats for different time granularity. For example, in a quarter-year chart, we may wish to see the month name with the first date of the month, while see the date name with others. This can be made with:
> ```ts
> formatter: {
>    month: '{MMMM}', // Jan, Feb, ...
>    day: '{d}' // 1, 2, ...
>}
>```

I guess that means this could actually be typed better:
```suggestion
    | { [key: CascadingTemplateLevels]: string };
```
where `CascadingTemplateLevels` is defined as:
```ts
type CascadingTemplateLevels =
  | 'year'
  | 'month'
  | 'day'
  | 'hour'
  | 'minute'
  | 'second'
  | 'millisecond'
  | 'none';
```

However, I think given your point in [this comment](https://github.com/mongodb/leafygreen-ui/pull/2536/files#r1826051813), I'm wondering if we should publicly support this. Given that EChart's string interpolation isn't standard JS either, I'm almost wondering if we should just publicly support the callback function. Consumers can still get hacky, override the type and use either of these, but by not publicly supporting them it probably helps save some headaches in the future if we ever need to switch the underlying library. What do you think?

Reply:
Decided to make the decision to only support the callback function and have pushed changes to reflect that. Let me know if there's any disagreement here.